Assets/HoloToolkit/Utilities/Scripts/HeadsUpDirectionIndicator.cs

```diff
@@ -0,0 +1,140 @@
+﻿using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    // The easiest way to use this script is to drop in the HeadsUpDirectionIndicator prefab
+    // from the HoloToolKit. If you're having issues with the prefab or can't find it,
+    // you can simply create a UI->Image object and attach this script on the Image child
+    // of the Canvas. In that case you'll need to pick your own sprite. You'll also need to
+    // choose an appropriate "pivot" point on the RectTransform. After that you simply
+    // need to specify the "targetObject" and then you should be set.
+    // 
+    // This script takes for granted that your UI is rendering in Camera space and that the
+    // up vector is aligned with the "pointing" direction of your image.
+    [RequireComponent(typeof(RectTransform))]
+    public class HeadsUpDirectionIndicator : MonoBehaviour
+    {
+        [Tooltip("The object the direction indicator will point to.")]
+        [SerializeField]
+        private GameObject targetObject;
+
+        [Tooltip("Determines what percentage of the visible field should be margin.")]
+        [Range(0.0f, 100.0f)]
+        [SerializeField]
+        private float indicatorMarginPercent;
+
+        private RectTransform rectTransform;
+
+        private const string MissingRectTransformErrorMessage = "The use of the HeadsUpDirectionIndicator component requires a RectTransform attached to the same game object.";
+
+        private void Start()
+        {
+            this.rectTransform = this.GetComponent<RectTransform>();
+            
+            if (this.rectTransform == null)
+            {
+                this.gameObject.SetActive(false);
+                Debug.LogError(MissingRectTransformErrorMessage);
+            }
+        }
+
+        // Update the direction indicator's position and orientation every frame.
+        private void Update()
+        {
+            if (targetObject == null)
```

Comment:
I think this should stay in update, so we can turn on and off our target indicator if we don't have a current TargetObject.